virtio-devices/src/balloon.rs

```diff
@@ -330,77 +331,84 @@ impl BalloonEpollHandler {
 }
 
 impl EpollHelperHandler for BalloonEpollHandler {
-    fn handle_event(&mut self, _helper: &mut EpollHelper, event: &epoll::Event) -> bool {
+    fn handle_event(
+        &mut self,
+        _helper: &mut EpollHelper,
+        event: &epoll::Event,
+    ) -> result::Result<(), EpollHelperError> {
         let ev_type = event.data as u16;
         match ev_type {
             RESIZE_EVENT => {
-                if let Err(e) = self.resize_receiver.evt.read() {
-                    error!("Failed to get resize event: {:?}", e);
-                    return true;
-                }
-                let mut signal_error = false;
-                let r = {
-                    let mut config = self.config.lock().unwrap();
-                    config.num_pages =
-                        (self.resize_receiver.get_size() >> VIRTIO_BALLOON_PFN_SHIFT) as u32;
-                    if let Err(e) = self.signal(VirtioInterruptType::Config) {
-                        signal_error = true;
-                        Err(e)
-                    } else {
-                        Ok(())
-                    }
-                };
-                if let Err(e) = &r {
-                    // This error will send back to resize caller.
-                    error!("Handle resize event get error: {:?}", e);
-                }
-                if let Err(e) = self.resize_receiver.send(r) {
-                    error!("Sending \"resize\" generated error: {:?}", e);
-                    return true;
-                }
-                if signal_error {
-                    return true;
-                }
```

Comment:
Could you please not try to improve the logic here but simply convert the code. I think we need the improvement to happen in a dedicated commit.